docs/rst/tutorials/micro/user_api/user_api_middleware.rst

```diff
@@ -0,0 +1,279 @@
+    rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+    rcl_context_t context = rcl_get_zero_initialized_context();
+    rcl_init_options_init(&init_options, rcl_get_default_allocator());
+
+    // Take RMW options from RCL options
+    rmw_init_options_t* rmw_options = rcl_init_options_get_rmw_init_options(&init_options);
+
+    // Set RMW client key
+    uint32_t client_key = 0xBA5EBA11;
+    rmw_uros_options_set_client_key(0xBA5EBA11, rmw_options);
+
+This feature can be useful for reusing DDS entities already created on the agent side, as explained on this `Micro XRCE-DDS Deployment example <https://micro-xrce-dds.docs.eprosima.com/en/latest/getting_started.html#deployment-example>`_.
+
+Time configuration
+------------------
+
+Timeouts
+^^^^^^^^
+
+- Confirmation timeout: Reliable entities can increase blocking time on ``rcl_publish`` or executor spin calls as they will wait for the acknowledgement of each sent message.
+
+  The default timeout value can be modified individually at run-time:
+
+  .. code-block:: c
+
+    // Confirmation timeout in milliseconds
+    int ack_timeout = 1000;
+
+    // Get RWM publisher handle and set reliable timeout
+    rmw_publisher_t* rmw_publisher_handle = rcl_publisher_get_rmw_handle(&publisher);
+    rmw_uros_set_publisher_session_timeout(&rmw_publisher_handle, ack_timeout);
+
+    // Get RWM service handle and set reliable timeout
+    rmw_service_t* rmw_service_handle = rcl_service_get_rmw_handle(&service);
+    rmw_uros_set_service_session_timeout(rmw_service_handle, ack_timeout);
+
+    // Get RWM service handle and set reliable timeout
+    rmw_client_t* rmw_client_handle = rcl_client_get_rmw_handle(&client);
+    rmw_uros_set_client_session_timeout(rmw_client_handle, ack_timeout);
+
+- Entity timeouts: Creation and destruction of entities also include a timeout, as they will wait for the Agent confirmation on the operation. This timeout will affect all ``init`` and ``fini`` methods such as ``rclc_node_init_default``, ``rcl_publisher_fini``, etc.
+
+  The default value can be modified at run-time for all entities:
+
+  .. code-block:: c
+
+      rclc_support_t support;
+      support.context
+
+      rmw_context_t * rmw_context = rcl_context_get_rmw_context(&support.context);
+
+      if (NULL != rmw_context)
+      {
+          // Timeout in milliseconds
+          int timeout = 1000;
+          rmw_uros_set_context_entity_creation_session_timeout(rmw_context, timeout);
+          rmw_uros_set_context_entity_destroy_session_timeout(rmw_context, timeout);
+      }
+
+
+  .. note::
+
+    To avoid waiting for agent confirmation ``timeout = 0`` can be used, allowing the release of local resources if the agent is not present.
+
+Time synchronization
+^^^^^^^^^^^^^^^^^^^^
+
+micro-ROS Clients can synchronize their epoch time with the connected Agent, this can be very useful when working in embedded environments that do not provide any time synchronization mechanism.
```

Comment:
Clients in lower case

Reply:
https://github.com/eProsima/vulcanexus/pull/78/commits/346a5be6837282f7b23d35cfe1375a8f7aa19ecd